dbms/src/Flash/Statistics/ExecutorStatisticsCollector.cpp

```diff
@@ -65,22 +80,121 @@ void ExecutorStatisticsCollector::initialize(DAGContext * dag_context_)
                 TableScanStatistics,
                 TopNStatistics,
                 WindowStatistics,
-                ExpandStatistics>(executor_id, &executor))
+                ExpandStatistics>(&executor))
         {
             throw TiFlashException(
-                fmt::format("Unknown executor type, executor_id: {}", executor_id),
+                fmt::format("Unknown executor type, executor_id: {}", executor.executor_id()),
                 Errors::Coprocessor::Internal);
         }
         return true;
     });
+
+    fillListBasedExecutorsChild();
+}
+
+void ExecutorStatisticsCollector::fillListBasedExecutorsChild()
+{
+    if (!dag_context->return_executor_id)
```

Comment:
`return_executor_id` seems to be a bad name at the moment.
- For tree base executor, there must be executor_id
- For list base executor, there must be no executor_id
In fact `is_list_base_executor` or `executor_dag_type` would be better.

Reply:
No, must  follow the logic in DAGContext.cpp:117.
```
{
    RUNTIME_CHECK(dag_request->has_root_executor() && dag_request->root_executor().has_executor_id());
    return_executor_id = dag_request->root_executor().has_executor_id() || dag_request->executors(0).has_executor_id();

    initOutputInfo();
    initListBasedExecutors();
}
```